ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/ExecutableServer.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"; you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.rest.server;
+
+import io.confluent.rest.ApplicationServer;
+import io.confluent.rest.RestConfig;
+import java.util.List;
+
+/**
+ * An {@code Executable} that wraps a {@link ApplicationServer} and delegates
+ * the lifecycle methods.
+ *
+ * @param <T> the type of the rest server config
+ */
+public class ExecutableServer<T extends RestConfig> implements Executable {
+
+  private final ApplicationServer<T> server;
+  private final List<ExecutableApplication<T>> apps;
+
+  public ExecutableServer(ApplicationServer<T> server, List<ExecutableApplication<T>> apps) {
+    this.server = server;
+    this.apps = apps;
+  }
+
+  @Override
+  public void startAsync() throws Exception {
+    apps.forEach(server::registerApplication);
+    server.start();
+
+    for (ExecutableApplication<T> app : apps) {
```

Comment:
Is having a list for the # of apps just in case later we decide to have multiple applications run on the server?

Reply:
yeah, this was the pattern that @rnpridgeon recommended. Maybe it's overkill for now... might be good if we want to model connect the same way (and perhaps other embedded apps)